lib/blumquist.rb

```diff
@@ -5,11 +5,15 @@
 require 'blumquist/errors'
 
 class Blumquist
+  attr_reader :_type
+
   def initialize(options)
     # Poor man's deep clone: json 🆗 🆒
     @data = JSON.parse(options.fetch(:data).to_json)
     @schema = options.fetch(:schema).with_indifferent_access
+    @original_properties = options.fetch(:schema).with_indifferent_access[:properties]
     @validate = options.fetch(:validate, true)
+    @_type = 'object'
```

Comment:
Perhaps then that `type` would be better than the inferred one, anyways :) But I get your point